From ab93845dd6b707082ba55060644c994dcc3ac94d Mon Sep 17 00:00:00 2001
From: Michael Adam <obnox@samba.org>
Date: Sat, 27 Feb 2016 14:02:02 +0100
Subject: [PATCH 1063/1096] smbd:smb2: add some asserts before decrementing the
 counters

Signed-off-by: Michael Adam <obnox@samba.org>
Reviewed-by: Jeremy Allison <jra@samba.org>
(cherry picked from commit f81f3a2d78832258b09bcc63d5cce2b4594cbbc8)

BUG: https://bugzilla.samba.org/show_bug.cgi?id=11809
(cherry picked from commit 15700e9ebe5cdc75ee7f7dca261a9b9c12589645)
---
 source3/smbd/smb2_server.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/source3/smbd/smb2_server.c b/source3/smbd/smb2_server.c
index ae125df..8a4aa96 100644
--- a/source3/smbd/smb2_server.c
+++ b/source3/smbd/smb2_server.c
@@ -2704,8 +2704,10 @@ static void smbd_smb2_request_reply_update_counts(struct smbd_smb2_request *req)
 	channel_sequence = SVAL(inhdr, SMB2_HDR_CHANNEL_SEQUENCE);
 
 	if (op->global->channel_sequence == channel_sequence) {
+		SMB_ASSERT(op->request_count > 0);
 		op->request_count -= 1;
 	} else {
+		SMB_ASSERT(op->pre_request_count > 0);
 		op->pre_request_count -= 1;
 	}
 }
-- 
2.9.0

